### PR TITLE
cleanup: spell DoH with a lowercase o

### PR DESCRIPTION
**Pull request by jsoref**
https://en.wikipedia.org/wiki/DNS_over_HTTPS

https://github.com/curl/curl/blob/3085ccfae996bb0fa606d2c7bc6783dc15d76a30/lib/doh.h#L31

I'm not entirely certain about the full set of changes, as some might change the behavior of clients that are parsing output. Happy to drop any sets.

### CODE REVIEW

File: docs/libcurl/opts/CURLOPT_DOH_SSL_VERIFYPEER.3 [hunk 1]
```diff
@@ -65,7 +65,7 @@ the correct end-point.
 .SH DEFAULT
 1
 .SH PROTOCOLS
-DOH
+DoH
```
Comment (jsoref) [inline docs/libcurl/opts/CURLOPT_DOH_SSL_VERIFYPEER.3:68]: Less certain about this

File: docs/libcurl/opts/CURLOPT_DOH_SSL_VERIFYSTATUS.3 [hunk 2]
```diff
@@ -22,27 +22,27 @@
...
+This option is the DoH equivalent of \fICURLOPT_SSL_VERIFYSTATUS(3)\fP and
+only affects requests to the DoH server.
 
 Note that if this option is enabled but the server does not support the TLS
 extension, the verification will fail.
 .SH DEFAULT
 0
 .SH PROTOCOLS
-DOH
+DoH
```
Comment (jsoref) [inline docs/libcurl/opts/CURLOPT_DOH_SSL_VERIFYSTATUS.3:45]: likewise

File: lib/doh.c [hunk 3]
```diff
@@ -186,19 +186,19 @@ doh_write_cb(const void *contents, size_t size, size_t nmemb, void *userp)
...
   struct Curl_easy *data = doh->set.dohfor;
   struct dohdata *dohp = data->req.doh;
-  /* so one of the DOH request done for the 'data' transfer is now complete! */
+  /* so one of the DoH request done for the 'data' transfer is now complete! */
   dohp->pending--;
-  infof(data, "a DOH request is completed, %u to go", dohp->pending);
+  infof(data, "a DoH request is completed, %u to go", dohp->pending);
   if(result)
-    infof(data, "DOH request %s", curl_easy_strerror(result));
+    infof(data, "DoH request %s", curl_easy_strerror(result));
```
Comment (jsoref) [inline lib/doh.c:198]: this pair changes output

File: lib/doh.c [hunk 4]
```diff
@@ -768,7 +768,7 @@ static void showdoh(struct Curl_easy *data,
   for(i = 0; i < d->numaddr; i++) {
     const struct dohaddr *a = &d->addr[i];
     if(a->type == DNS_TYPE_A) {
-      infof(data, "DOH A: %u.%u.%u.%u",
+      infof(data, "DoH A: %u.%u.%u.%u",
```
Comment (jsoref) [inline lib/doh.c:771]: likewise

File: lib/doh.c [hunk 5]
```diff
@@ -777,7 +777,7 @@ static void showdoh(struct Curl_easy *data,
       char buffer[128];
       char *ptr;
       size_t len;
-      msnprintf(buffer, 128, "DOH AAAA: ");
+      msnprintf(buffer, 128, "DoH AAAA: ");
```
Comment (jsoref) [inline lib/doh.c:780]: likewise

File: lib/doh.c [hunk 6]
```diff
@@ -958,7 +958,7 @@ CURLcode Curl_doh_is_resolved(struct Curl_easy *data,
                             &de);
       Curl_dyn_free(&p->serverdoh);
       if(rc[slot]) {
-        infof(data, "DOH: %s type %s for %s", doh_strerror(rc[slot]),
+        infof(data, "DoH: %s type %s for %s", doh_strerror(rc[slot]),
```
Comment (jsoref) [inline lib/doh.c:961]: likewise

File: lib/doh.c [hunk 7]
```diff
@@ -969,7 +969,7 @@ CURLcode Curl_doh_is_resolved(struct Curl_easy *data,
       struct Curl_dns_entry *dns;
       struct Curl_addrinfo *ai;
 
-      infof(data, "DOH Host name: %s", dohp->host);
+      infof(data, "DoH Host name: %s", dohp->host);
```
Comment (jsoref) [inline lib/doh.c:972]: likewise

**Review by jsoref — COMMENTED** [review]
These are the bits that seem the most risky

**Review by bagder — APPROVED** [review]
I think this is generally good since everyone and the RFC use `DoH` these days. Originally the protocol was not written like that and that's the reason it snuck into curl with the uppercase o...

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**